### PR TITLE
Fix: should decode a protocol message that has an error without a message

### PR DESCRIPTION
**Pull request by ricardopereira**
#589

### CODE REVIEW

**Comment by mattheworiordan** [comment]
@ricardopereira thanks for this, however I am not sure this quite addresses the problem correctly. It appears that the library currently expects a JSON / MsgPack response in all cases. However, if for example one of our ELBs returns an error, it will typically not even have a content type and be a simple text response.  Can you add a test that makes sure the library copes with an error message in plaintext and HTML format from ELB?

**Comment by mattheworiordan** [comment]
PS. If the response from the server is not of the correct format, it would be useful if you could artificially construct an ErrorInfo object with the message being the response body (perhaps limit to 1kb in case a large HTML page response is returned) and the `statusCode` property populated with the HTTP status code.

File: Spec/RestClient.swift [hunk 1]
```diff
@@ -1322,6 +1322,19 @@ class RestClient: QuickSpec {
                 }
             }
 
+            // https://github.com/ably/ably-ios/issues/589
+            it("client should handle error messages in plaintext and HTML format") {
```
Comment (ricardopereira) [inline Spec/RestClient.swift]: @mattheworiordan PTAL

(BTW, this will fail. Just ensuring this is enough.)

Comment (mattheworiordan) [inline Spec/RestClient.swift]: Sure, test looks good, just need the code now :)

**Comment by mattheworiordan** [comment]
@ricardopereira thanks, I see the test are still failing though. Are those failing tests related to this issue?

**Comment by ricardopereira** [comment]
@tcard Still failing. Should I handle this?

File: Source/ARTRest.m [hunk 2]
```diff
@@ -213,31 +213,27 @@ - (void)executeRequest:(NSURLRequest *)request completion:(void (^)(NSHTTPURLRes
...
-            NSError *dataError = [self->_encoders[response.MIMEType] decodeError:data];
-            if ([self shouldRenewToken:&dataError]) {
-                [self.logger debug:__FILE__ line:__LINE__ message:@"RS:%p retry request %@", self, request];
-                // Make a single attempt to reissue the token and resend the request
-                if (_tokenErrorRetries < 1) {
-                    [self executeRequest:request withAuthOption:ARTAuthenticationTokenRetry completion:callback];
-                    return;
+            if (data) {
+                NSError *dataError = [self->_encoders[response.MIMEType] decodeError:data];
+                if ([self shouldRenewToken:&dataError]) {
```
Comment (ricardopereira) [inline Source/ARTRest.m]: @tcard I think you're missing the else case?

Comment (tcard) [inline Source/ARTRest.m]: Right, fixed at c595705.

Comment (ricardopereira) [inline Source/ARTRest.m]: @tcard the `shouldRenewToken` changes the `dataError` but the right now it ignores that change.

**Comment by tcard** [comment]
Test are passing except for a few that have nothing to do with this. @ricardopereira can you do a release for 0.8 and 1.0?

**Comment by ricardopereira** [comment]
Yes, sure. But I'm AFTK now.

**Pull request merged**